main.tf

```diff
@@ -23,36 +23,40 @@ resource "null_resource" "shell" {
   }
 
   provisioner "local-exec" {
-    command = "${local.command_chomped} 2>\"${path.module}/stderr.${null_resource.start.id}\" >\"${path.module}/stdout.${null_resource.start.id}\"; echo $? >\"${path.module}/exitstatus.${null_resource.start.id}\""
+    command     = "${local.command_chomped} 2>\"${abspath(path.module)}/stderr.${null_resource.start.id}\" >\"${abspath(path.module)}/stdout.${null_resource.start.id}\"; echo $? >\"${abspath(path.module)}/exitstatus.${null_resource.start.id}\""
```

Comment:
what does `abspath` change? why is it required?

Reply:
Without `abspath` the stderr, stdout, and exitstatus files will be created in the child module's directory. Normally that's fine, but that causes problems with the external data sources. 

I parameterized module_path as a local so you can see what I'm talking about. I think things go sideways without `abspath` only if you specify the working_dir

Reply:
@userhas404d can you give me an concrete example of "Normally that's fine, but that causes problems with the external data sources." because I just don't fully understand this